app/controllers/issues_controller.rb

```diff
@@ -135,6 +127,20 @@ def liquid_resource_assigns
     { 'issue' => IssueDrop.new(@issue) }
   end
 
+  def redirect_to_main_or_qa
+    notice = 'Issue updated.'
+
+    if params[:return_to] == 'qa'
+      if @issue.ready_for_review?
+        redirect_to_target_or_default project_qa_issue_path(current_project, @issue), notice: notice
```

Comment:
Do we need to use `redirect_to_target_or_default` here since we're no longer setting `return_to` in the session? What about using a regular `redirect_to`? (same on L 137 & 140)